File: Problemas/110A - Nearly Lucky Number/110A_Nearly_Lucky_Number.cpp
// 110A - Nearly Lucky Number
// https://codeforces.com/problemset/problem/110/A

#include <stdio.h>
#include <stdlib.h>
#include <bits/stdc++.h>
using namespace std;

int main(){

    // Número a checar los dígitos
    long long int numero;
    cin >> numero;

    // Contador de dígitos de la suerte
    int contador_suerte = 0;

    // Mientras tu numero sea positivo
    while(numero > 0) {

        // Realiza módulo 10 para sacar el último dígito del número
        // Si el dígito es 4 o 7, añade 1 al contador
        if(numero % 10 == 4 || numero % 10 == 7) {
            contador_suerte++;
        }

        // Divide el número entre 10 para eliminar el último dígito
        numero = numero / 10;
    }

    // Si el número tiene 4 o 7 dígitos de la suerte, entonces imprime 'YES', de lo contrario, imprime 'NO'
    if(contador_suerte == 4 || contador_suerte == 7) {
        cout << "YES" << endl;
    } else {
        cout << "NO" << endl;
    }

    return 0;
}

File: Problemas/110A - Nearly Lucky Number/110A_Nearly_Lucky_Number.py
# 110A - Nearly Lucky Number
# https://codeforces.com/problemset/problem/110/A

# Número a checar los dígitos
numero = int(input())

# Convierte el número a string
numero_string = str(numero)

# Contador de dígitos de la suerte
contador_suerte = 0

# Checa cada dígito del número
for digito in numero_string:

    # Si el dígito es 4 o 7, añade 1 al contador
    if digito == '4' or digito == '7':
        contador_suerte += 1

# Si el número tiene 4 o 7 dígitos de la suerte, entonces imprime 'YES', de lo contrario, imprime 'NO'
if contador_suerte == 4 or contador_suerte == 7:
    print('YES')
else:
    print('NO')

